### PR TITLE
[Merged by Bors] - feat: charP_iff_prime_eq_zero

### PR DESCRIPTION
**Pull request by ChrisHughes24**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/CharP/Basic.lean [hunk 1]
```diff
@@ -560,7 +560,20 @@ end Semiring
...
+theorem charP_iff_eq_zero_of_prime {p : ℕ} (hp : p.Prime) : CharP R p ↔ (p : R) = 0 :=
+  ⟨fun h => cast_eq_zero R p, by
+    rw [charP_iff]
+    intro hp0 n
+    refine ⟨?_, fun ⟨_, hpn⟩ => by simp_all⟩
+    rw [hp.dvd_iff_not_coprime, Nat.coprime_iff_gcd_eq_one,
+      ← Nat.cast_inj (R := ℤ), Nat.gcd_eq_gcd_ab]
+    intro hn0 hgcd
+    apply_fun ((↑) : ℤ → R) at hgcd
+    simp [hp0, hn0] at hgcd⟩
```
Comment (eric-wieser) [inline Mathlib/Algebra/CharP/Basic.lean:574]: Is there a faster route here via `ringChar_of_prime_eq_zero` (which is incorrectly in the `CharOne` section below)

Comment (ChrisHughes24) [reply Mathlib/Algebra/CharP/Basic.lean]: Done and I also renamed the section

File: Mathlib/Algebra/CharP/Basic.lean [hunk 2]
```diff
@@ -605,7 +605,12 @@ theorem ringChar_of_prime_eq_zero [Nontrivial R] {p : ℕ} (hprime : Nat.Prime p
   Or.resolve_left ((Nat.dvd_prime hprime).1 (ringChar.dvd hp0)) ringChar_ne_one
 #align char_p.ring_char_of_prime_eq_zero CharP.ringChar_of_prime_eq_zero
 
-end CharOne
+theorem charP_iff_prime_eq_zero [Nontrivial R] {p : ℕ} (hp : p.Prime) :
```
Comment (eric-wieser) [inline Mathlib/Algebra/CharP/Basic.lean:608]: I'd be tempted to add a `cast` to this name to match `cast_eq_zero`, though you should do the same to the `ringChar` lemma above (and the PR title) if you agree:
```suggestion
theorem charP_iff_cast_eq_zero_of_prime [Nontrivial R] {p : ℕ} (hp : p.Prime) :
```
bors d+

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**